### PR TITLE
add contralateral referencing example

### PR DESCRIPTION
**Pull request by drammock**
this adds a new example, based on this forum post:

https://mne.discourse.group/t/referencing-to-contralateral-mastoids/6587/6

### CODE REVIEW

File: examples/preprocessing/contralateral_referencing.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+                               .drop_channels(mastoids)
+                            )
+
+# contralateral referencing (alters channels in `raw` in-place):
+for ref, hemi in dict(M2=ch_names['Left'], M1=ch_names['Right']).items():
+    mne.set_bipolar_reference(
+        raw, anode=hemi, cathode=[ref] * len(hemi), copy=False
+    )
+# strip off '-M1' and '-M2' suffixes added to each bipolar-referenced channel
+raw.rename_channels(lambda ch_name: ch_name.split('-')[0])
```
Comment (hoechenberger) [inline examples/preprocessing/contralateral_referencing.py:63]: `set_bipolar_reference()` has a `ch_name` parameter for this purpose, maybe we can use it to simplify things?

File: examples/preprocessing/contralateral_referencing.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    'TP9': 'M1',
+    'TP10': 'M2'
+})
+
+# this splits electrodes into 3 groups; left, midline, and right
+ch_indices = mne.channels.make_1020_channel_selections(raw.info)
+
+# convert indices to names
+orig_names = np.array(raw.ch_names)
+ch_names = {key: orig_names[idxs].tolist() for key, idxs in ch_indices.items()}
```
Comment (hoechenberger) [inline examples/preprocessing/contralateral_referencing.py:38]: Feature request: Allow retrieval of channel names (not only indices) via `make_1020_channel_selections()`, e.g. by adding a new parameter `return_ch_names`

File: examples/preprocessing/contralateral_referencing.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+# to the mean of the two mastoid channels; the left and right hemispheres we'll
+# reference to the single contralateral mastoid channel.
+
+# midline referencing to mean of mastoids:
+mastoids = ['M1', 'M2']
+rereferenced_midline_chs = (raw.copy()
+                               .pick(mastoids + ch_names['Midline'])
+                               .set_eeg_reference(mastoids)
+                               .drop_channels(mastoids)
+                            )
```
Comment (hoechenberger) [inline examples/preprocessing/contralateral_referencing.py:55]: blackify if we want to go that route?

```suggestion
rereferenced_midline_chs = (
    raw.copy()
    .pick(mastoids + ch_names['Midline'])
    .set_eeg_reference(mastoids)
    .drop_channels(mastoids)
)
```

**Review by hoechenberger — COMMENTED** [review]
Nice, thanks for the effort, @drammock!

File: examples/preprocessing/contralateral_referencing.py (hunk 2 above)
Comment (drammock) [reply examples/preprocessing/contralateral_referencing.py:38]: https://github.com/mne-tools/mne-python/issues/11597

File: examples/preprocessing/contralateral_referencing.py (hunk 1 above)
Comment (drammock) [reply examples/preprocessing/contralateral_referencing.py:63]: not actually possible to just set the original `anode` names:

```py
In [1]: mne.set_bipolar_reference(raw, anode=ch_names['Left'], cathode=['M2'] * len(ch_
   ...: names['Left']), ch_name=ch_names['Left'])
ValueError: The following channels are present in more than one input measurement info objects: ['C3', 'P3', 'Fp1', 'FC1', 'F7', 'PO9', 'FC5', 'F3', 'P7', 'O1', 'T7', 'CP1', 'CP5']
```

File: examples/preprocessing/contralateral_referencing.py (hunk 3 above)
Comment (drammock) [reply examples/preprocessing/contralateral_referencing.py:55]: not worth the extra CI cycles IMO. Plan is to blackify whole codebase before next release, so not important to do it now.

**Comment by drammock** [comment]
rendered doc looks good: https://output.circle-artifacts.com/output/job/a08d726b-cb9c-4862-9ebf-fe4953587fb5/artifacts/0/dev/auto_examples/preprocessing/contralateral_referencing.html

ready for merge IMO.

**Review by agramfort — APPROVED** [review]

File: examples/preprocessing/contralateral_referencing.py (hunk 1 above)
Comment (hoechenberger) [reply examples/preprocessing/contralateral_referencing.py:63]: Bummer! Thanks for trying

**Comment by hoechenberger** [comment]
Thanks @drammock!

**Pull request merged**